datafusion-federation/src/lib.rs

```diff
@@ -16,9 +16,15 @@ use datafusion::{
 };
 
 pub use optimizer::{get_table_source, FederationOptimizerRule};
-pub use plan_node::{FederatedPlanNode, FederatedQueryPlanner, FederationPlanner};
+pub use plan_node::{
+    FederatedPlanNode, FederatedPlanner, FederatedQueryPlanner, FederationPlanner,
+};
 pub use table_provider::{FederatedTableProviderAdaptor, FederatedTableSource};
 
+// TODO clean up this
+// TODO move schema_cast.rs to schema_cast directory
+pub mod schema_cast;
```

Comment:
This schema_cast module is useful when the schema registered in DataFusion and the schema returned by the underlying provider don't match exactly. The schema_cast provides a mechanism to keep the schema aligned with what DataFusion expects.

Reply:
I put `TODO` as reminder there to clean up the module by moving schema_cast.rs to schema_cast/mod.rs